File: backend-1/services/RabbitMQService.js
import amqp from "amqplib"

class RabbitMQService{
    constructor(){
        this.connection = null,
        this.channel = null
        this.username = process.env.RABBITMQ_USERNAME
        this.password = process.env.RABBITMQ_PASSWORD
    }

    async connect(){
        try {
            this.connection = await amqp.connect(`amqp://${this.username}:${this.password}@localhost`)
            this.channel = await this.connection.createChannel();
            console.log('RabbitMQ connected in Backend 1')
        } catch (error) {
            console.log('RabbitMQ connection error:', error)
        }
    }

    async createQueue(queueName){
        if(!this.channel) throw new Error("Channel not initialized")
        await this.channel.assertQueue(queueName,{durable:true})
    }

    async closeConnection() {
        try {
          if (this.channel) {
            await channel.close();
            console.log('Channel closed');
          }
          if (this.connection) {
            await connection.close();
            console.log('Connection closed');
          }
        } catch (error) {
          console.error('Error closing connection:', error);
        }
    }

    getChannel() {
        return this.channel;
    }
}

const rabbitMQService = new RabbitMQService()

export default rabbitMQService;

File: backend-2/api/services/RabbitMQService.py
import pika
import os
from dotenv import load_dotenv

load_dotenv()
class RabbitMQService:
    def __init__(self):
        self.connection = None
        self.channel = None
        self.username = os.getenv("RABBITMQ_USERNAME")
        self.password =os.getenv("RABBITMQ_PASSWORD")

    def connect(self):
        try:
            
            if not self.username or not self.password:
                raise ValueError("RabbitMQ username or password is missing from environment variables.")
            
            self.connection = pika.BlockingConnection(
                pika.ConnectionParameters(host='localhost', credentials=pika.PlainCredentials(self.username,self.password))
            )
            self.channel = self.connection.channel()
            print("RabbitMQ connected in Backend 2")
        except Exception as e:
            print(f"Error connecting to RabbitMQ: {e}")

    def create_queue(self, queue_name):
        if not self.channel:
            raise Exception("Channel not initialized")
        self.channel.queue_declare(queue=queue_name, durable=True)

    def get_channel(self):
        return self.channel
